### DISCUSSION TITLE
How to setup up 'client_max_body_size' in this proxy? (upload large files)

### DISCUSSION THREAD

**Discussion by cod3r0k** [evertramos/nginx-proxy-automation#261] [Q&A] [answered]
# Error description:
gitlab error 413 (big file can't be uploaded). How can we setup nginx to upload large file?

**Comment by evertramos** [comment]
might be... enable the nginx conf and set your max file upload. make sure you have the same settings in your local application set up.

**Comment by gustawdaniel** [comment]
This is third issue in this topic:

https://github.com/evertramos/docker-compose-letsencrypt-nginx-proxy-companion/issues/207
https://github.com/evertramos/docker-compose-letsencrypt-nginx-proxy-companion/issues/81

I have this problem too some times. We should add chapter in documentation that explains possible 
solutions and best practices of solving this problem.

**Comment by evertramos** [comment, marked as answer]
Thanks @gustawdaniel for your suggestion! Let's see if we make this clear and help others.

This proxy allows you to set _'client_max_body_size'_ in couple ways. 

1. Use `$ ./fresh-start.sh` with `--use-nginx-conf-files' argument

This way you will have 100m allowed as of [uploadsize.conf](https://github.com/evertramos/nginx-proxy-automation/blob/master/conf.d/uploadsize.conf)

2. If you already have the proxy running add [this file](https://github.com/evertramos/nginx-proxy-automation/blob/master/conf.d/uploadsize.conf) to _`./data/conf.d`_ folder.

3. You can set this option per site as of [nginx-proxy doc](https://github.com/nginx-proxy/nginx-proxy)

- [Per-host](https://github.com/nginx-proxy/nginx-proxy#per-virtual_host)
- [Per-host location](https://github.com/nginx-proxy/nginx-proxy#per-virtual_host-location-configuration)

Adding this setting per site you must use the site (url) name as mentioned in [this sample](https://github.com/evertramos/nginx-proxy-automation/issues/262#issuecomment-812901849) as well.

You can set _client_max_body_size_ at multiple context as of [nginx docs](http://nginx.org/en/docs/http/ngx_http_core_module.html#client_max_body_size), you can use it per _location_ if you with to or at _vhost (http)_ level.

So, if your url is _`my-very-own-domain.com`, and you intend to use at _host_ level you create a file named **my-very-own-domain.com** at your _'./data/vhost.d'_ path, you may find the right path at [NGINX_FILES_PATH variable at .env file](https://github.com/evertramos/nginx-proxy-automation/blob/master/.env.sample#L57).

Your file will look like:

```nginx
client_max_body_size 1024m;
```

Then you must stop your site container (example the nextcloud) and start again, so the _nginx-proxy_ will update the options or if you can not stop it, you can reload _nginx_ options using the following:

```bash
$ docker exec -it [YOUR_NGINX_PROXY_CONTAINER_NAME] nginx -s reload
```

Please let me know if this was useful.

Thanks!

**Reply by Pyroseza** [reply]
Thanks for showing me this discussion Evert, apologies for raising the issue earlier

**Reply by evertramos** [reply]
I just replied, right after your issue @Pyroseza, so thank you for raising up this subject again so I could stop and answer it the _right_ way!

**Reply by ivanhercaz** [reply]
Hi @evertramos! First of all, thank you very much for your awesome work with `nginx-proxy-automation`. I am having a similar issue but I don't if you prefer that I open a new thread or to expose my problem here so, what do you prefer?

And yes, I tried to reproduce the same steps, but it works with one domain and not with another one.
